superset/assets/javascripts/explore/components/ExploreViewContainer.jsx

```diff
@@ -101,13 +106,43 @@ class ExploreViewContainer extends React.Component {
     }
   }
 
+  addHistory(isReplace = false, title) {
+    const { payload } = createExplore(this.props.form_data);
+    const longUrl = getExploreLongUrl(this.props.form_data);
+    if (isReplace) {
+      history.replaceState(
+        payload,
+        document.title,
+        longUrl);
+    } else {
+      history.pushState(
+        payload,
+        document.title,
+        longUrl);
+    }
+
+    // it seems browsers don't support pushState title attribute
+    if (title) {
+      document.title = title;
+    }
+  }
+
   handleResize() {
     clearTimeout(this.resizeTimer);
     this.resizeTimer = setTimeout(() => {
       this.setState({ height: this.getHeight(), width: this.getWidth() });
     }, 250);
   }
 
+  handleNavigate() {
+    const formData = history.state;
+    if (formData && Object.keys(formData).length) {
+      this.props.actions.setExploreControls(formData);
+      this.props.actions.runQuery(formData, false,
```

Comment:
nit -- I would do newlines for each `var` here.

Reply:
fixed.